### ISSUE TITLE
Reverse direction  of `From` of what your package does

### ISSUE THREAD

**Issue by vporton** [bobozaur/transitive#1]
I need the reverse direction  of `From` of what your package does:

```rust
use derive_more::From;

// Suppose `HttpStatus` is declared in a library that we can't modify.
// We don't have `transitive_from` here.
struct HttpStatus {
  code: u8,
}

#[derive(From)]
enum TransportError {
  Status(Status),
}

#[transitive_to(HttpStatus, Error, TransportError)] // this?
#[transitive_to(TransportError, Error, HttpStatus)] // or this?
// Should define
// impl From<HttpStatus> for Error;
// You `transitive_from` can define only
// impl From<Error> for HttpStatus; // wrong
#[derive(From)]
enum Error {
  Transport(TransportError),
}
```

Such error conversions abound in my code and I really need this.

**Comment by bobozaur** [comment]
Interesting idea.

I think this might actually play even better with `TransitiveTryFrom` as it's quite non-intuitive that it transitions in reverse.
Introducing `to` and `from` might make this more powerful.

I'll toy around with it and see how to accommodate this.

**Comment by bobozaur** [comment]
@vporton I published version 0.2.0 which allows the reverse functionality. It might take some time before the docs are up, so feel free to look at the doc tests in in the [lib.rs](https://github.com/bobozaur/transitive/blob/main/src/lib.rs) or at the integration tests.

For consistency, the crate now exposes `TransitiveInto`, `TransitiveTryInto`, `TransitiveFrom` and `TransitiveTryFrom`.

`TransitiveInto` is the previous `TransitiveFrom` (the name makes more sense given the conversion direction), `TransitiveTryFrom`'s behavior remains unchanged (allowing reverse impls similar to what you had in mind, but fallible) and the other two derive macros are their counterparts.

I'd appreciate if you got back with a comment on this issue confirming whether this accommodates your use case :)!

Thanks again for the suggestion!

**Comment by bobozaur** [comment]
It looks like I rushed this a bit.

The problem with the approach above is that, because all derive macros use the same attribute, you cannot provide different lists for different derives. The original idea with the `to` and `from` arguments was better, and I'll implement that in a new minor release.

Closing in favor of #2.
